From d8cd411b904e11831808d123a4426645523b3c68 Mon Sep 17 00:00:00 2001
From: David Sniatkiwicz <davidsn@marvell.com>
Date: Sun, 29 Oct 2017 14:58:31 +0200
Subject: [PATCH 2042/2241] fix: dts: a8k-ocp: fix USB Host0 on CP1 for OCP
 board

Currently, USB vBus disconnected during linux boot on OCP board.
This patch fixes this issue by changing the USB Host0 on CP1 node
to USB2 with UTMI only.

Change-Id: I4d061276f1e0f64cbacac36bba354de70e185afb
Signed-off-by: David Sniatkiwicz <davidsn@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/45649
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-ocp.dts | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts b/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
index 985f876..5a0416b 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
@@ -156,18 +156,20 @@
 			cps_comphy: comphy {
 				status = "okay";
 			};
+			/* cps_usb3h0 is USB2 only (UTMI) */
 			cps_usb3h0: usb3@500000 {
 				usb-phy = <&cps_usb3h0_phy>;
-				separated-phys-for-usb2-usb3;
-				phys = <&cps_utmi0>,
-				       <&cps_comphy 2 COMPHY_USB3H0>;
-				phy-names = "usb2", "usb3";
+				phys = <&cps_utmi0>;
+				phy-names = "usb2";
 				status = "okay";
 			};
 			cps_usb3h0_phy: usb3_phy0 {
 				compatible = "usb-nop-xceiv";
 				vcc-supply = <&reg_usb3h0_vbus>;
 			};
+			cps_utmi0: utmi@580000 {
+				status = "okay";
+			};
 			gop {
 				cps_emac0: mac0 {
 					phys = <&cps_comphy 4 COMPHY_SFI>;
-- 
2.7.4

